Add default docs folder structure (#1)
iota-wiki/docs-template: documentation/README.md

# Documentation

The documentation is built using [Docusaurus 2](https://docusaurus.io/). The deployment is done through a centralized build from [IOTA WIKI](https://github.com/iota-community/iota-wiki). To run a local instance the [IOTA WIKI CLI](https://github.com/iota-community/iota-wiki-cli) is used.

## Prerequisites

- [Node.js v14.14+](https://nodejs.org/en/)
- [yarn](https://yarnpkg.com/getting-started/install)

## Installation

```console
yarn
```

> This command installs all needed dependencies.

Reviewer: ```suggestion
This command installs all necessary dependencies.
```

## Local Development

```console
yarn start
```

> This command starts a local, wiki themed, development server and opens up a browser window. Most changes are reflected live without having to restart the server.

Reviewer: ```suggestion
This command starts a local, wiki themed development server and opens up a browser window. Most changes are reflected live without having to restart the server.
```

## Including .md file

```console
{@import <file path>}
```

Example:

```console
{@import ../../../../bindings/wasm/docs/api-reference.md}

> ```

Reviewer 2: I'm actually not a fan of this parse tree manipulation using remark plugins for cases like this. I rather have either plain Markdown or [the MDX functionality provided by Docusaurus](https://docusaurus.io/docs/markdown-features/react#importing-markdown) where plain Markdown proves to be insufficient.

````suggestion
## Including Markdown or MDX from another file
*partial.mdx*

```
<span>This is some {props.text}.</span>

And plain **markdown** is also supported.
```

*page.mdx*

```
import Partial from './partial.md'

<Partial text='partial markdown' />
```

Would render:
> This is some partial markdown.
> 
> And plain **markdown** is also supported.
````

Author: I must say I agree, this file is copy-pasted from another project.  I thought perhaps you and Kevin didn't want to complicate users with some React.  Though to be honest it's pretty much the same except the imports are scattered all over the place with one method, which usually is hard to maintain.

Reviewer: I actually also just used it everywhere because identity had it in there and other projects used it.
My plan was to eventually remove it anyway and add all this stuff to the wiki as a guide. So we could link our devs there if they want to see what is possible and what we allow.
Same for default plugins which are available for all repos.

Also not using react came mainly from the editor. But I guess we just drop that project :trollface: 
But tbh we always agreed that we should allow more freedom in the external repo docs

## Default Structure

IOTA projects usually house their Wiki documentation within the documentation folder of any project. This
template provides a set structure to separate concerns based on the [Diataxis framework](https://diataxis.fr/), with
some minor modifications.

You do not need to use all of these sections, so please feel free to delete any which do not suit your needs.

### Getting Started

> The [Getting Started folder](documentation/getting_started/README.md) should be the first section in your documentation. It should give the user a high level overview of the project, required prior knowledge, prerequisites, and ideally a quick set up guide, or "hello world".

Reviewer 3: ```suggestion
The [Getting Started folder](documentation/getting_started/README.md) should be the first section in your documentation. It should give the user a high-level overview of the project, required prior knowledge, prerequisites, and ideally a quick setup guide, or "hello world".
```

> ### Key Concepts

Reviewer: Do we also want to keep It in this order?
Is key concepts abstracted enough? If it is more theory we should move it down in the sidebar as it won't be read by many people 🤔

Reviewer 3: Maybe the  `Examples` & `Tutorials` need to move to the top?

The [Key concepts folder](documentation/key_concepts/README.md) revolves around [explanations](https://diataxis.fr/explanation/), and are therefore **understanding oriented**.

### Examples

The [Examples folder](documentation/examples/README.md) should only address concrete examples, or [how-to guides](https://diataxis.fr/how-to-guides/), which are **goal oriented**.

### Reference

The [Reference folder](documentation/reference/README.md) should describe the [technical information](https://diataxis.fr/reference/) of your project. It is **information-oriented**.

### Tutorials

The [Tutorials folder](documentation/tutorials/README.md) should contain articles which guide the user step by step through a series of how-tos with the relevant explanations to achieve a project or real world use case. [Tutorials](https://diataxis.fr/tutorials/) are **learning-oriented**.

> ### Troubleshooting

Reviewer: I guess this could then also contain FAQs if needed?

# Documentation

The documentation is built using [Docusaurus 2](https://docusaurus.io/). The deployment is done through a centralized build from [IOTA WIKI](https://github.com/iota-community/iota-wiki). To run a local instance the [IOTA WIKI CLI](https://github.com/iota-community/iota-wiki-cli) is used.

## Prerequisites

- [Node.js v14.14+](https://nodejs.org/en/)
- [yarn](https://yarnpkg.com/getting-started/install)

## Installation

```console
yarn
```

This command installs all necessary dependencies.

## Local Development

```console
yarn start
```

This command starts a local, wiki themed development server and opens up a browser window. Most changes are reflected live without having to restart the server.

> ## Including Markdown or MDX from another file

Reviewer: So I guess we would remove this and replace it with something like:
```
You can find guidance and an introduction to the needed tools and syntax on [our wiki](https://wiki.iota.org/participate/contribute-to-wiki/for_devs/developer_guide)
```

And there we could potentially also describe the structure below, but mostly tools like the cli, diataxis, docusaurus tools and more 🤔

> ## Including Markdown or MDX from another file
> *partial.mdx*
> 
> ```
> <span>This is some {props.text}.</span>
> 
> And plain **markdown** is also supported.
> ```
> 
> *page.mdx*
> 
> ```
> import Partial from './partial.md'
> 
> <Partial text='partial markdown' />
> ```
> 
> Would render:
> > This is some partial markdown.
> > 
> > And plain **markdown** is also supported.

Author: ```suggestion
## MDX Features

You can find guidance and an introduction to the needed tools and syntax on [our wiki](https://wiki.iota.org/participate/contribute-to-wiki/for_devs/developer_guide)

```

This command starts a local, wiki themed development server and opens up a browser window. Most changes are reflected live without having to restart the server.

> ## MDX Features

Reviewer: Can we have another title for this?

Something broader like Features and tools?

Because we may also explain the cli, the plugin system, default plugins, and so on 🤔

Author: ```suggestion
## Features And Tools
```
